### ISSUE TITLE
fix diffs

### ISSUE THREAD

**Issue by ppelleti** [haskell/haskell-wiki-configuration#1]
Currently, diffing two revisions of a page does not work (diff is empty).

**Comment by ppelleti** [comment]
I've [posted a question][1] on StackExchange, but no answer yet.

[1]: https://superuser.com/questions/1141625/mediawiki-extension-wikidiff2-not-working-diffs-are-empty

**Comment by gbaz** [comment]
There might be something suggestive in the error/web logs when such a case
is hit?

**Comment by ppelleti** [comment]
Perhaps, but if so, I haven't been able to find the proper log yet.  It's not showing up in `/var/log/nginx/error.log` or `/var/log/php5-fpm.log`.  There isn't anything else that seems promising in `/var/log`, and if there's a MediaWiki-specific log, I haven't found it yet.

However, there is some promising information at https://www.mediawiki.org/wiki/Manual:How_to_debug about enabling logging, so I will look into that further.

**Comment by ppelleti** [comment]
I enabled logging by adding:

    $wgDebugLogFile = "/var/log/mediawiki/debug.log";

to `/etc/mediawiki/LocalSettings.php`.  I then discovered this in the log file:

```
wfShellExec: 'wikidiff2' '/home/web/wikidata/upload/tmp/diff_f9tuWC' '/home/web/wikidata/upload/tmp/diff_TsMWXK'
Possibly missing executable file: 'wikidiff2' '/home/web/wikidata/upload/tmp/diff_f9tuWC' '/home/web/wikidata/upload/tmp/diff_TsMWXK'
```

This suggests that for some reason it is trying to run an executable named `wikidiff2`, rather than using the `wikidiff2` extension.  Now I just need to figure out why...

**Comment by ppelleti** [comment]
This is partially fixed now, but I'm leaving the bug open.

I fixed it by putting:

    $wgExternalDiffEngine = false;

in `LocalSettings.php`.  I hadn't tried it before because this is supposed to be the default, and it wasn't being set anywhere in `LocalSettings.php`.  Apparently, it must have been set somewhere else.

Now, we do get diffs, which is much better than before, when we were getting nothing.  Now, the diffs look like this:

![diffs-without-coloring](https://cloud.githubusercontent.com/assets/1080008/19984594/567c4226-a1cc-11e6-9d57-eccf6a121ead.png)

However, the diffs look better on Wikipedia, where they are colorized:

![diffs-with-coloring](https://cloud.githubusercontent.com/assets/1080008/19984653/97f22dd8-a1cc-11e6-8bbb-55b085de1aa6.png)

All the necessary information is there in the HTML, so it's just a matter of applying the right styles in the CSS.  So, I suspect that the Hawiki theme needs to be modified in order to color the diffs.

I'll also bet the diffs would be colorized in the MonoBook theme, but in trying that out, I ran into #5.
